shared_workflow/shared_automated_workflow.py

```diff
@@ -8,101 +8,60 @@
 import sys
 from datetime import datetime
 from logging import Logger
-from subprocess import Popen, PIPE
 from typing import List
 
 import qcore.constants as const
-from qcore.config import host
 from qcore.utils import load_yaml
 from scripts.management.MgmtDB import MgmtDB
 from qcore.qclogging import get_basic_logger, NOPRINTCRITICAL
+from scripts.schedulers.scheduler_factory import get_scheduler
 
 ALL = "ALL"
 ONCE = "ONCE"
 ONCE_PATTERN = "%_REL01"
 NONE = "NONE"
 
 
-def get_queued_tasks(user=None, machine=const.HPC.maui):
-    if user is not None:
-        cmd = "squeue -A {} -o '%A %t' -M {} -u {}".format(
-            const.DEFAULT_ACCOUNT, machine.value, user
-        )
-    else:
-        cmd = "squeue -A {} -o '%A %t' -M {}".format(
-            const.DEFAULT_ACCOUNT, machine.value
-        )
-    process = Popen(shlex.split(cmd), stdout=PIPE, encoding="utf-8")
-    (output, err) = process.communicate()
-    process.wait()
-    try:
-        header = output.split("\n")[1]
-    except:
-        raise EnvironmentError(
-            "squeue did not return expected output. Ignoring for this iteration. Actual output: {}".format(
-                output
-            )
-        )
-    else:
-        if header != "JOBID ST":
-            raise EnvironmentError(
-                "squeue did not return expected output. Ignoring for this iteration."
-            )
-    output_list = list(filter(None, output.split("\n")[1:]))
+def get_queued_tasks(user=None, machine=None):
+    scheduler = get_scheduler()
+    output_list = scheduler.check_queues(user=user, target_machine=machine)
     return output_list
 
 
 def submit_sl_script(
```

Comment:
now this function is not slurm specific, should we change the function name? generate_sl_script() might be also changed.